clients/Android/ultrablue/app/src/main/java/fr/gouv/ssi/ultrablue/model/UltrablueProtocol.kt

```diff
@@ -48,15 +72,58 @@ class UltrablueProtocol(private val activity: MainActivity, private var device:
                 device.ekn = key.N
                 device.eke = key.E.toInt()
                 device.ekcert = key.Cert
+                logger?.push(Log("Registering device"))
                 activity.viewModel.insert(device)
+                logger?.push(CLog("Device has been registered", true))
                 writeMsg("registration confirmation", byteArrayOf(-10)) // Registration success, CBOR encoded
             }
             AUTHENTICATION_READ -> readMsg("authentication nonce")
             AUTHENTICATION -> {
-                // When encryption will be implemented, we'll first need to decrypt the nonce here.
-                writeMsg("Nonce", message)
+                val authNonce = Cbor.decodeFromByteArray<ByteArrayModel>(message)
+                //TODO: When encryption will be implemented, we'll need to decrypt the nonce here.
+                val encodedAuthNonce = Cbor.encodeToByteArray(authNonce)
+                writeMsg("Nonce", encodedAuthNonce)
             }
             AK_READ -> readMsg("attestation key")
+            CREDENTIAL_ACTIVATION -> {
+                encodedAttestationKey = message
+                logger?.push(Log("Generating credential challenge"))
+                val credentialBlob = Gomobile.makeCredential(device.ekn, device.eke.toLong(), encodedAttestationKey)
```

Comment:
makeCredential returns an invalid blob in case of error. We should check it here.

Maybe better, raise an exception instead and catch it?